distributed-counter/src/main/java/org/infinispan/counter/CounterEvent.java

```diff
@@ -0,0 +1,22 @@
+package org.infinispan.counter;
+
+/**
+ * The event used by {@link CounterListener}.
+ *
+ * @author Pedro Ruivo
+ * @since 9.0
+ */
+public interface CounterEvent {
+
+   /**
+    * @return The counter value.
+    */
+   long getValue();
```

Comment:
Is that the before- or after- value? (the same for state) Can't we have both of them?